### PR TITLE
[DPE-6035] Clear ops warnings

### PR DESCRIPTION
**Pull request by dragomirp**
* Clear OPS import warnings
* Update DPW
* Jimmy-rig the test app build

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -80,9 +82,9 @@ jobs:
       - lint
       - unit-test
       - build
-    uses: canonical/data-platform-workflows/.github/workflows/integration_test_charm.yaml@v23.0.4
+    uses: canonical/data-platform-workflows/.github/workflows/integration_test_charm.yaml@v23.0.5
     with:
-      artifact-prefix: packed-charm-cache-true
+      artifact-prefix: packed-charm-cache
```
Comment (dragomirp) [inline .github/workflows/ci.yaml:87]: Conditional cache

File: src/charm.py [hunk 2]
```diff
@@ -21,10 +21,18 @@
...
+    ActiveStatus,
+    BlockedStatus,
+    CharmBase,
+    EventBase,
+    Relation,
+    RelationBrokenEvent,
+    RelationEvent,
+    StatusBase,
+    main,
+)
```
Comment (dragomirp) [inline src/charm.py:35]: OPS made most imports direct.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -53,14 +53,16 @@ jobs:
         charm:
           - path: .
             charmcraft: 2.x/stable
+            cache: true
           - path: tests/integration/app-charm
             charmcraft: latest/stable
+            cache: false
```
Comment (dragomirp) [inline .github/workflows/ci.yaml:59]: Test app builds on charmcraft 3, due to strict binary dependencies, but that breaks charmcraftcache-hub

**Review by welpaolo — APPROVED** [review]
LGTM

**Pull request merged**